sdk/typespec/typespec_derive/src/lib.rs

```diff
@@ -0,0 +1,205 @@
+extern crate proc_macro;
+
+use proc_macro2::{Span, TokenStream};
+use syn::parse::ParseStream;
+use syn::spanned::Spanned;
+use syn::{Attribute, DeriveInput, LitStr, Meta, Path};
+
+/// Derive macro for implementing `Model` trait.
+/// Deriving this trait allows a type to be deserialized from an HTTP response body.
+/// By default, the type must also implement `serde::Deserialize`, or the generated code will not compile.
```

Comment:
My idea for a derive was to actually do this, or do we want to push any customization of `#[serde(...)]` attributes to explicit decisions? I was going to allow it anyway, but maybe making devs explicitly derive `Deserialize` (and/or `Serialize`, which you may also want to mention) makes it more obvious of what's expected. Thoughts?